### PR TITLE
Connect redux with saga to users

### PR DESCRIPTION
**Pull request by krakamod**
Create actions, reducers, sagas and connect all this staff with users (main page, where displayed users table)

### CODE REVIEW

File: src/actions/server.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// @flow
+import {
+  SERVER_REQUESTED,
+  SERVER_REQUEST_SUCCEEDED,
+  SERVER_REQUEST_FAILED,
+} from '../types/actionTypes';
+
+type Request = { +type: typeof SERVER_REQUESTED };
+type RequestSuccess = { +type: typeof SERVER_REQUEST_SUCCEEDED };
```
Comment (vik-buchinski) [inline src/actions/server.js]: remove SERVER_REQUESTED and SERVER_REQUEST_SUCCEEDED

File: src/containers/users/index.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+// @flow
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router-dom';
+import { requestUsers } from '../../actions/users';
+import Users from './users';
+
+const mapStateToProps = ({ users, server }) => ({
```
Comment (vik-buchinski) [inline src/containers/users/index.js]: please add
`const mapStateToProps = ({ users, server }: State) => ({`

File: src/actions/users.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+// @flow
+import { USERS_REQUESTED, USERS_REQUEST_SUCCEEDED } from '../types/actionTypes';
+import { type User } from '../types/userDefs';
+
+type SuccessPayload = {
+  +users: $ReadOnlyArray<User>,
```
Comment (vik-buchinski) [inline src/actions/users.js]: 👍

File: src/reducers/server.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  error: '',
+  loading: false,
+};
+
+const serverReducer = (state: State = initialState, action: Actions): State => {
+  switch (action.type) {
+    case SERVER_REQUESTED:
+      return { error: '', loading: true };
+    case SERVER_REQUEST_SUCCEEDED:
+      return { error: '', loading: false };
```
[the remark this answers is not in the document]
Comment (krakamod) [reply src/reducers/server.js]: I don't like this, but ok

**Pull request merged**